### PR TITLE
add log replay comparisons in fake cloud logging server

### PR DESCRIPTION
**Pull request by Ruide**
a follow up on previous [PR #417](https://github.com/google/go-tpm-tools/pull/417)

### CODE REVIEW

**Comment by Ruide** [comment]
Hi @alexmwu ! Could you take a look?

File: cmd/fake_cloudlogging_server.go [hunk 1]
```diff
@@ -120,6 +121,29 @@ func (h *loggingHandler) WriteLogEntries(_ context.Context, req *logpb.WriteLogE
 		// Store by log name.
 		h.logs[e.LogName] = append(h.logs[e.LogName], e)
 	}
+
+	var logEntryPayload []map[string]interface{}
+	logEntryPayload = append(logEntryPayload, map[string]interface{}{"aud": "test", "iat": float64(1709752525), "exp": float64(1919752525)})
+	logEntryPayload = append(logEntryPayload, map[string]interface{}{"token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJ0ZXN0IiwiaWF0IjoxNzA5NzUyNTI1LCJleHAiOjE5MTk3NTI1MjV9.EBLA2zX3c-Fu0l--J9Gey6LIXMO1TFRCoe3bzuPGc1k"})
+	logEntryPayload = append(logEntryPayload, map[string]interface{}{"Name": "projects/test-project/locations/us-central-1/challenges/947b4f7b-e6d4-4cfe-971c-39ffe00268ba", "Nonce": "GoogAttestV1xkIPiQz1O8T_O88A4cv4iA", "ConnID": ""})
+	logEntryPayload = append(logEntryPayload, map[string]interface{}{"Name": "projects/test-project/locations/us-central-1/challenges/947b4f7b-e6d4-4cfe-971c-39ffe00268ba", "Nonce": "R29vZ0F0dGVzdFYxeGtJUGlRejFPOFRfTzg4QTRjdjRpQQ==", "ConnID": ""})
```
Comment (alexmwu) [inline cmd/fake_cloudlogging_server.go:129]: nit: the nonce value and challenge UUIDs should be constants

File: cmd/fake_cloudlogging_server.go [hunk 2]
```diff
@@ -120,6 +121,29 @@ func (h *loggingHandler) WriteLogEntries(_ context.Context, req *logpb.WriteLogE
 		// Store by log name.
 		h.logs[e.LogName] = append(h.logs[e.LogName], e)
 	}
+
+	var logEntryPayload []map[string]interface{}
+	logEntryPayload = append(logEntryPayload, map[string]interface{}{"aud": "test", "iat": float64(1709752525), "exp": float64(1919752525)})
+	logEntryPayload = append(logEntryPayload, map[string]interface{}{"token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJ0ZXN0IiwiaWF0IjoxNzA5NzUyNTI1LCJleHAiOjE5MTk3NTI1MjV9.EBLA2zX3c-Fu0l--J9Gey6LIXMO1TFRCoe3bzuPGc1k"})
+	logEntryPayload = append(logEntryPayload, map[string]interface{}{"Name": "projects/test-project/locations/us-central-1/challenges/947b4f7b-e6d4-4cfe-971c-39ffe00268ba", "Nonce": "GoogAttestV1xkIPiQz1O8T_O88A4cv4iA", "ConnID": ""})
```
Comment (alexmwu) [inline cmd/fake_cloudlogging_server.go:128]: Why are there two nonce logs?

Comment (Ruide) [reply cmd/fake_cloudlogging_server.go]: One is redundant. Removed.

File: cmd/fake_cloudlogging_server.go (hunk 1 above)
Comment (Ruide) [reply cmd/fake_cloudlogging_server.go]: Fixed!

**Review by alexmwu — APPROVED** [review]

**Pull request merged**